src/Riak/Api/Http.php

```diff
@@ -718,30 +719,27 @@ protected function parseResponse()
                 break;
 
             case 'Basho\Riak\Command\MapReduce\Fetch':
-                $results = null;
-                if ($body) {
-                    $results = json_decode($body);
-                }
+                $results = in_array($this->statusCode, [200,204]) ? json_decode($body) : null;
                 $response = new Command\MapReduce\Response($this->success, $this->statusCode, $this->error, $results);
                 break;
             case 'Basho\Riak\Command\Indexes\Query':
-                $body = json_decode(rawurldecode($body), true);
+                $json_object = in_array($this->statusCode, [200,204]) ? json_decode($body, true) : null;
                 $results = [];
                 $termsReturned = false;
                 $continuation = null;
                 $done = true;
 
-                if (isset($body['keys'])) {
-                    $results = $body['keys'];
+                if (isset($json_object['keys'])) {
```

Comment:
@christophermancini I would say do it only if you think it will add clarity and value to the code. What I wanted to say is that I think that it shouldn't say object when it's an array instead.